### ISSUE TITLE
Rule proposal: `no-useless-ternary`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1798]
### Description

I was looking for a rule to forbid ternary that equivalent to a "logical or" expression.

The only thing I got is [`no-unneeded-ternary`](https://eslint.org/docs/rules/no-unneeded-ternary) rule, it checks `x ? x : 1` behind an option. [Even if I set `defaultAssignment: false`, this rule only check identifier](https://eslint.org/docs/rules/no-unneeded-ternary#defaultassignment)

### Fail

```js
foo ? foo : bar;
```

```js
foo.bar ? foo.bar : foo.baz
```

### Pass

```js
foo || bar;
```

```js
foo.bar || foo.baz;
```

### Additional Info

We should check all expressions instead of identifiers only.

This rule is safe to fix to logical or(`||`), but sometimes `??` operator maybe better.

So, we may want to use the suggestions API to draw attention?

**Comment by fisker** [comment]
### fail

```js
!bar ? foo : bar;
```

**Comment by B2o5T** [comment]
another case
```js
foo?.bar ? foo.bar : qux
```

**Comment by fisker** [comment]
Should we name it `prefer-logical-operator` or `prefer-logical-operator-over-tenery` to be more specific?
